server/entitymanager/entitymanager.go

```diff
@@ -2,38 +2,154 @@
+func (m *InMemoryEntityManager) Sign(resp *bootz.GetBootstrapDataResponse, serial string) error {
+	// Sign the response
+	block, _ := pem.Decode([]byte(m.artifacts.OC.Key))
+	if block == nil {
+		return status.Errorf(codes.Internal, "unable to decode OC private key")
+	}
+	priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return err
+	}
+	if resp.GetSignedResponse() == nil {
+		return status.Errorf(codes.InvalidArgument, "empty signed response")
+	}
+	signedResponseBytes, err := proto.Marshal(resp.GetSignedResponse())
+	if err != nil {
+		return err
+	}
+	hashed := sha256.Sum256(signedResponseBytes)
+	sig, err := rsa.SignPKCS1v15(nil, priv, crypto.SHA256, hashed[:])
+	if err != nil {
+		return err
+	}
+	resp.ResponseSignature = string(sig)
+	// Populate the OV
+	ov, err := m.FetchOwnershipVoucher(serial)
+	if err != nil {
+		return err
+	}
+	resp.OwnershipVoucher = []byte(ov)
+	// Populate the OC
+	resp.OwnershipCertificate = []byte(m.artifacts.OC.Cert)
+	return nil
 }
 
-// GetBootstrapData returns the Bootstrap data for the provided control card.
-func (m *InMemoryEntityManager) GetBootstrapData(*bootz.ControlCard) (*bootz.BootstrapDataResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "Unimplemented")
+// FetchOwnershipVoucher retrieves the ownership voucher for a control card
+func (m *InMemoryEntityManager) FetchOwnershipVoucher(serial string) (string, error) {
+	if ov, ok := m.artifacts.OV[serial]; ok {
+		return ov, nil
+	}
+	return "", status.Errorf(codes.NotFound, "OV for serial %v not found", serial)
 }
 
-// SetStatus returns the current status based on the status request.
-func (m *InMemoryEntityManager) SetStatus(*bootz.ReportStatusRequest) error {
-	return status.Errorf(codes.Unimplemented, "Unimplemented")
+// AddControlCard adds a new control card to the entity manager.
+func (m *InMemoryEntityManager) AddControlCard(serial string) *InMemoryEntityManager {
+	m.controlCardStatuses[serial] = bootz.ControlCardState_CONTROL_CARD_STATUS_UNSPECIFIED
+	return m
 }
 
-// Sign populates the signing fields of the provided Bootstrap data response.
-// If fields are set they will be overwritten.
-func (m *InMemoryEntityManager) Sign(*bootz.GetBootstrapDataResponse) error {
-	return status.Errorf(codes.Unimplemented, "Unimplemented")
+// AddChassis adds a new chassis to the entity manager.
+func (m *InMemoryEntityManager) AddChassis(bootMode bootz.BootMode, manufacturer string, serial string) *InMemoryEntityManager {
+	l := service.EntityLookup{
+		Manufacturer: manufacturer,
+		SerialNumber: serial,
+	}
+	e := service.ChassisEntity{
+		BootMode: bootMode,
+	}
+	m.chassisInventory[l] = &e
+	return m
 }
 
 // New returns a new in-memory entity manager.
-func New() *InMemoryEntityManager {
-	return &InMemoryEntityManager{}
+func New(artifacts *service.SecurityArtifacts) *InMemoryEntityManager {
```

Comment:
we should have a clear workflow on how artifacts get loaded into the entity manager - i am seeing AddChasssis and AddControlCard